src/org/gk/qualityCheck/QACheckUtilities.java

```diff
@@ -26,4 +33,297 @@ public static GKInstance getLatestCuratorIEFromInstance(GKInstance instance) thr
+        Set<GKInstance> physicalEntities = new HashSet<>();
+        if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
+            // Recursively search Complex components for all PhysicalEntities.
+            Collection<GKInstance> complexComponents = multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.hasComponent);
+            for (GKInstance complexComponent : complexComponents) {
+                physicalEntities.addAll(findAllPhysicalEntities(complexComponent));
+            }
+        } else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
+            // Recursively search Polymer repeatedUnits for all PhysicalEntities.
+            Collection<GKInstance> polymerUnits = multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.repeatedUnit);
+            for (GKInstance polymerUnit : polymerUnits) {
+                physicalEntities.addAll(findAllPhysicalEntities(polymerUnit));
+            }
+        } else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
+            // Recursively search members and candidates (if EntitySet is a CandidateSet) for all PhysicalEntities.
+            physicalEntities.addAll(findEntitySetPhysicalEntities(multiPEInstance));
+        }
+        return physicalEntities;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist within an EntitySet by searching for all member PEs (via 'hasMember' attribute)
+     * and all candidate PEs (if entitySet is a CandidateSet; via 'hasCandidate attribute).
+     * @param entitySet GKInstance -- EntitySet instance that will be searched for all PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in EntitySet.
+     * @throws Exception -- Thrown by MySQLAdaptor.
+     */
+    public static Set<GKInstance> findEntitySetPhysicalEntities(GKInstance entitySet) throws Exception {
+        Set<GKInstance> physicalEntities = new HashSet<>();
+        // Attribute values to be recursively searched for PhysicalEntities include 'hasMember' and 'hasCandidate'.
+        Set<String> entitySetAttributes = new HashSet<>(Arrays.asList(ReactomeJavaConstants.hasMember, ReactomeJavaConstants.hasCandidate));
+        // If the EntitySet is a DefinedSet, 'hasCandidate' attribute is not searched.
+        if (entitySet.getSchemClass().isa(ReactomeJavaConstants.DefinedSet)) {
+            entitySetAttributes.remove(ReactomeJavaConstants.hasCandidate);
+        }
+        for (String entitySetAttribute : entitySetAttributes) {
+            // Recursively searches members/candidates in EntitySet for all PhysicalEntities.
+            for (GKInstance setInstance : (Collection<GKInstance>) entitySet.getAttributeValuesList(entitySetAttribute)) {
+                physicalEntities.addAll(findAllPhysicalEntities((setInstance)));
+            }
+        }
+        return physicalEntities;
+    }
+
+    // Returns true if 'inferredFrom' referral is null.
```

Comment:
Comment should be formatted as Javadoc. Also for `isHumanDatabaseObject`, `hasSpeciesAttribute`, `getInstanceAttributeName`; `setSkipList` & `setHumanSpeciesInst` are missing Javadoc.